client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,279 @@
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+import notices from 'notices';
+import protectForm from 'lib/mixins/protect-form';
+import FormInput from 'components/forms/form-text-input-with-affixes';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import CountedTextarea from 'components/forms/counted-textarea';
+import analytics from 'lib/analytics';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+
+export default React.createClass( {
+	displayName: 'SiteSettingsFormSEO',
+
+	mixins: [ protectForm.mixin ],
+
+	getInitialState() {
+		return this.stateForSite( this.props.site );
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( get( nextProps, 'site.ID', 0 ) !== get( this.props, 'site.ID', 0 ) ) {
+			if ( get( nextProps, 'site.jetpack' ) ) {
+				// Go back to general settings if switched to a Jetpack site
+				page( this.getGeneralSettingsUrl( get( nextProps, 'site.slug', '' ) ) );
+				return;
+			}
+
+			// Update state when switching sites
+			this.setState( this.stateForSite( nextProps.site ) );
+		}
+	},
+
+	stateForSite: function( site ) {
+		return {
+			seoMetaDescription: get( site, 'options.seo_meta_description', '' ),
+			verificationServicesCodes: {
+				google: get( site, 'options.verification_services_codes.google', '' ),
+				bing: get( site, 'options.verification_services_codes.bing', '' ),
+				pinterest: get( site, 'options.verification_services_codes.pinterest', '' ),
+				yandex: get( site, 'options.verification_services_codes.yandex', '' )
+			}
+		};
+	},
+
+	handleMetaChange( event ) {
+		this.setState( { seoMetaDescription: get( event, 'target.value', '' ) } );
+	},
+
+	handleVerificationCodeChange( event, serviceName ) {
+		const servicesCodes = this.state.verificationServicesCodes;
+		if ( ! servicesCodes.hasOwnProperty( serviceName ) ) {
+			return;
+		}
+
+		servicesCodes[ serviceName ] = get( event, 'target.value', '' );
+
+		this.setState( { verificationServicesCodes: servicesCodes } );
+	},
+
+	submitSeoForm( event ) {
+		const site = this.props.site;
+
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		notices.clearNotices( 'notices' );
+
+		this.setState( { submittingForm: true } );
+
+		const filteredCodes = omitBy( this.state.verificationServicesCodes, isBoolean );
+		const updatedOptions = {
+			seo_meta_description: this.state.seoMetaDescription,
+			verification_services_codes: filteredCodes
+		};
+
+		site.saveSettings( updatedOptions, error => {
+			if ( error ) {
+				switch ( error.error ) {
+					case 'invalid_ip':
+						notices.error( this.translate( 'One of your IP Addresses was invalid. Please, try again.' ) );
```

Comment:
I don't know my way around the form, but would a user know what this message means? Are they entering an IP address? Is there a way to indicate what's wrong with it? Also, since this is a notice, it pops up at the top of the page. It might seem a bit disconnected from the form element where it belongs.